common/src/main/java/net/adoptopenjdk/icedteaweb/StringUtils.java

```diff
@@ -67,4 +69,36 @@ public static boolean isBlank(final String str) {
         lines.add(tmp);
         return lines;
     }
+
+    /**
+     * Checks whether the first part of the given prefixString is a prefix for any of the strings
+     * in the specified array.
+     * If the {@code prefixString} contains multiple words separated by a space character, the
+     * first word is taken as prefix for comparison.
+     *
+     * @param prefixString the prefixString string
+     * @param available the strings to test
+     * @return true if the first part of the given prefixString is a prefix for any of the strings
+     * in the specified array or the specified array is empty or null, false otherwise
+     */
+    public static boolean hasPrefixMatch(final String prefixString, final String[] available) {
+        Assert.requireNonBlank(prefixString, "prefixString");
+
+        if (available == null || available.length == 0){
+            return true;
+        }
+
+        for (final String candidate : available) {
+            final String trimmedPrefix = prefixString.split(WHITESPACE_CHARACTER_SEQUENCE)[0];
+            String trimmedCandidate = null;
```

Comment:
I believe the splitting of the candidates can be omitted.
Calling `candidate.startsWith(trimmedPrefix)` should yield the same result.

But before changing this we should add some more tests for corner cases.
- Strings which start with one or multiple spaces

Reply:
would not work, see last test I've added in prefixShouldNotBeIncludedInArrayOfStrings(). Also I would not spend to much in refactoring of the logic now, see my first comment above.